packages/core/admin/admin/src/pages/ProfilePage/index.js

```diff
@@ -1,160 +1,268 @@
-
-export default () => 'TODO profile page';
+import React from 'react';
+import {
+  CustomContentLayout,
+  Form,
+  GenericInput,
+  useAppInfos,
+  useFocusWhenNavigate,
+  useNotification,
+  useOverlayBlocker,
+  auth,
+} from '@strapi/helper-plugin';
+import { useIntl } from 'react-intl';
+import { Formik } from 'formik';
+import { useQuery, useMutation, useQueryClient } from 'react-query';
+import pick from 'lodash/pick';
+import omit from 'lodash/omit';
+import { Helmet } from 'react-helmet';
+import { Main } from '@strapi/parts/Main';
+import { H3 } from '@strapi/parts/Text';
+import { Box } from '@strapi/parts/Box';
+import { HeaderLayout } from '@strapi/parts/Layout';
+import { Button } from '@strapi/parts/Button';
+import { Grid, GridItem } from '@strapi/parts/Grid';
+import { Stack } from '@strapi/parts/Stack';
+import { Select, Option } from '@strapi/parts/Select';
+import CheckIcon from '@strapi/icons/CheckIcon';
+import useLocalesProvider from '../../components/LocalesProvider/useLocalesProvider';
+import { fetchUser, putUser } from './utils/api';
+import { schema, layout } from './utils';
+
+const ProfilePage = () => {
+  const { changeLocale, localeNames } = useLocalesProvider();
+  const { setUserDisplayName } = useAppInfos();
+  const queryClient = useQueryClient();
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { lockApp, unlockApp } = useOverlayBlocker();
+  useFocusWhenNavigate();
+
+  const { status, data } = useQuery('user', () => fetchUser(), {
+    onError: () => {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error', defaultMessage: 'An error occured' },
+      });
+    },
+  });
+
+  const isLoading = status !== 'success';
+
+  const submitMutation = useMutation(body => putUser(omit(body, 'confirmPassword')), {
+    onSuccess: async data => {
+      await queryClient.invalidateQueries('user');
+
+      auth.setUserInfo(data);
+      const userDisplayName = data.username || `${data.firstname} ${data.lastname}`;
+      setUserDisplayName(userDisplayName);
+      changeLocale(data.preferedLanguage);
+
+      toggleNotification({
+        type: 'success',
+        message: { id: 'notification.success.saved', defaultMessage: 'Saved' },
+      });
+
+      unlockApp();
+    },
+    onError: () => {
+      toggleNotification({
+        type: 'warning',
+        message: { id: 'notification.error', defaultMessage: 'An error occured' },
+      });
+      unlockApp();
+    },
+    refetchActive: true,
+  });
+
+  const { isLoading: isSubmittingForm } = submitMutation;
+
+  const handleSubmit = async body => {
+    lockApp();
+
+    const username = body.username || null;
+    submitMutation.mutate({ ...body, username });
+  };
+
+  const fieldsToPick = ['email', 'firstname', 'lastname', 'username', 'preferedLanguage'];
+
+  const initialData = Object.keys(pick(data, fieldsToPick)).reduce((acc, current) => {
```

Comment:
This is not needed you can just do `const initialData = pick(data, fieldsToPick);`